### PR TITLE
VSIX: Call Tree: Bold essential nodes.

### PR DESCRIPTION
**Pull request by rtaket**
- Bold essential nodes.
- Do not highlight rows in treeview.
- Remove red bang for essential nodes.
- Expand essential nodes.

treenodestyle

### CODE REVIEW

**Comment by rtaket** [comment]
I'm not sure I like this new style. Would appreciate opinions...

File: src/Sarif.Viewer.VisualStudio/Themes/CallTrees.xaml [hunk 1]
```diff
@@ -31,22 +31,23 @@
...
+                        </TreeView.ItemContainerStyle>
+
                         <TreeView.ItemTemplate>
                             <HierarchicalDataTemplate DataType="x:Type local:CallTreeNode" ItemsSource="{Binding Children}">
-                                <StackPanel Orientation="Horizontal">
-                                    <TextBlock Text="! " FontWeight="Bold" Foreground="Red"
-                                               Style="{StaticResource callTreeTextStyle}"
-                                               Background="{Binding Location.Importance, 
-                                                            Converter={StaticResource importanceToBackgroundConverter}}"
-                                               Visibility="{Binding Location.Importance,
```
Comment (lgolding) [inline src/Sarif.Viewer.VisualStudio/Themes/CallTrees.xaml]: If you're going to remove the background coloring, then remove the `importanceToBackgroundConverter` resource, the `ImportanceToBackgroundConverter` class, and its unit tests.

Comment (lgolding) [inline src/Sarif.Viewer.VisualStudio/Themes/CallTrees.xaml]: Likewise if you're removing the red-bang, then remove the `importanceToEssentialMarkerVisibilityConverter` resource, the `ImportanceToEssentialMarkerVisibilityClass`, and its unit tests.

File: src/Sarif.Viewer.VisualStudio/Themes/DefaultStyles.xaml [hunk 2]
```diff
@@ -49,6 +49,23 @@
                 Value="Courier New" />
         <Setter Property="FontSize"
                 Value="12" />
+        <Style.Triggers>
+            <DataTrigger Binding="{Binding Location.Importance}"
+                         Value="Essential">
+                <Setter Property="FontWeight"
+                        Value="Bold" />
```
Comment (lgolding) [inline src/Sarif.Viewer.VisualStudio/Themes/DefaultStyles.xaml]: Huh, for some reason I never think of using style triggers for things like this. I would've written an `ImportanceToFontWeightConverter`. Your way is much better.

**Comment by lgolding** [comment]
:+1: Subject to

1. @michaelcfanning weighing in on appearance.
2. Removal of now-unused resources, classes, and tests.

File: src/Sarif.Viewer.VisualStudio/Themes/DefaultStyles.xaml (hunk 2 above)
Comment (lgolding) [inline src/Sarif.Viewer.VisualStudio/Themes/DefaultStyles.xaml]: ... and for that matter you could get rid of the `ImportanceToForegroundConverter` and its unit tests, and use style triggers for that as well.

**Comment by jericahuang** [comment]
I've also been experimenting with @michaelcfanning 's styling suggestions. Here's what the exclamation point in a blue circle would look like...
bluebang

**Comment by rtaket** [comment]
Maybe we should steal the error list icons.

image

**Comment by lgolding** [comment]
Good so far. You can also remove the `ImportanceToEssentialMarkerVisibility` class, and its tests, and the resource.

**Comment by lgolding** [comment]
:+1: @michaelcfanning has yet to weigh in on styling. But with these changes, it's so trivial to update the styling that I'd say, go ahead and merge this, if only for the sake of getting rid of all that code.

**Pull request merged**